src/main/java/net/sf/jabref/logic/groups/GroupsParser.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2003-2016 JabRef contributors.
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+package net.sf.jabref.logic.groups;
+
+import java.util.List;
+
+/**
+ * Converts string representation of groups to a parsed {@link GroupTreeNode}.
+ */
+public class GroupsParser {
+
+    public static GroupTreeNode importGroups(List<String> orderedData) throws Exception {
+        GroupTreeNode cursor = null;
+        GroupTreeNode root = null;
+        for (String string : orderedData) {
+            // This allows to read databases that have been modified by, e.g., BibDesk
+            string = string.trim();
+            if (string.isEmpty()) {
+                continue;
+            }
+
+            int spaceIndex = string.indexOf(' ');
+            if (spaceIndex <= 0) {
+                throw new Exception("bad format");
```

Comment:
I introduced a new checked `ParseException` since the `IllegalFormatException` is "thrown when a format string contains an illegal syntax" according to the documentation (so the format string is illegal and not the string in an ill-formed format).